File: clean-episode-dates.py
import pandas as pd
import re

# Load the raw content of the file
file_path = 'Episode-Dates.csv'

with open(file_path, 'r') as file:
    raw_content = file.readlines()

# Process the raw content to extract the episode name, date, and extra information
processed_data = []
for line in raw_content:
    match = re.match(r'"(.+)"\s\((.+)\)', line.strip())
    if match:
        episode_name = match.group(1)
        date_extra = match.group(2)
        # Extract date and extra info
        date_match = re.match(r'([\w]+\s\d{1,2},\s\d{4})(.*)', date_extra)
        if date_match:
            date = date_match.group(1).strip()
            extra = date_match.group(2).strip().replace(')', '').replace('(', '')  # Remove all `(` and `)` characters
        else:
            date = date_extra.strip()
            extra = ""
        processed_data.append({"Episode Name": episode_name, "Date": date, "Extra": extra})

# Convert the processed data to a DataFrame
processed_df = pd.DataFrame(processed_data)

# Save the cleaned data to a new CSV file
output_file_path = 'Episode-Dates.csv'
processed_df.to_csv(output_file_path, index=False)

print(f"Cleaned data has been saved to {output_file_path}.")

File: clean-episode-dates.js
const fs = require("fs");
const { parse } = require("json2csv");

const filePath = "Episode-Dates.csv";

fs.readFile(filePath, "utf8", (err, data) => {
  if (err) {
    console.error("Error reading the file:", err);
    return;
  }

  const rawContent = data.split("\n");

  const processedData = rawContent
    .map((line) => {
      const match = line.trim().match(/"(.+)"\s\((.+)\)/);
      if (match) {
        let title = match[1];
        const dateExtra = match[2];

        title = title.replace("Mount McKinley", "Mt. McKinley");

        const dateMatch = dateExtra.match(/([\w]+\s\d{1,2},\s\d{4})(.*)/);
        let date, extra;

        if (dateMatch) {
          date = dateMatch[1].trim();
          extra = dateMatch[2].trim().replace(/[()]/g, "");
        } else {
          date = dateExtra.trim();
          extra = "";
        }

        return { Title: title, Date: date, Extra: extra };
      }
      return null;
    })
    .filter((item) => item !== null);

  const fields = ["Title", "Date", "Extra"];
  const csvData = parse(processedData, { fields });

  fs.writeFile(filePath, csvData, (err) => {
    if (err) {
      console.error("Error writing to the file:", err);
    } else {
      console.log(`Cleaned data has been saved to ${filePath}.`);
    }
  });
});
